sunpy/net/dataretriever/sources/bbso.py

```diff
@@ -0,0 +1,120 @@
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+from bs4 import BeautifulSoup
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+from sunpy.time import TimeRange
+
+__all__ = ['BBSOClient']
+
+class BBSOClient(GenericClient):
+    """
+    Returns a list of URLS to BBSO files corresponding to value of input timerange.
+    URL source: `http://www.bbso.njit.edu/pub/archive/`.
+
+
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'bbso'
+
+    Level: Level can take only 'fl' or 'fr' as arguments.
+
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+
+    >>> results = Fido.search(Time('2016/5/18 15:28:00','2016/5/18 16:30:00'), a.Instrument('bbso'), a.Level('fr'))
+    >>> print(results)
+    [<Table length=2>
+         Start Time           End Time              Source        Instrument
+           str19               str19                str21            str4   
+    ------------------- ------------------- --------------------- ----------
+    2016-05-18 00:00:00 2016-05-19 00:00:00 Global Halpha Network       bbso
+    2016-05-19 00:00:00 2016-05-20 00:00:00 Global Halpha Network       bbso]
```

Comment:
How come the range of the date is a full day?, and if you are asking for just data between an hour... how you get the data for the next day?